### PR TITLE
Added zhmc_session module.

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.
The motivation is to save HMC sessions to be created and deleted for each task, if a playbook has multiple tasks for the zHMC.

Since many files have been changed, some hints for what to review:
* The new zhmc_session.py module
* The new open_session(), close_session() functions in common.py
* One of the other modules that now take session_id as a credential (they all have the same changes)

### CODE REVIEW

File: docs/source/modules/zhmc_adapter.rst [hunk 1]
```diff
@@ -52,14 +52,14 @@ hmc_auth
   userid
     The userid (username) for authenticating with the HMC.
```
Comment (simon-spinner) [inline docs/source/modules/zhmc_adapter.rst:53]: To be precise you would also need to mention that it is required if session_id is not provided.

File: plugins/modules/zhmc_adapter.py [hunk 2]
```diff
@@ -867,10 +857,13 @@ def main():
             required=True,
             type='dict',
             options=dict(
-                userid=dict(required=True, type='str'),
-                password=dict(required=True, type='str', no_log=True),
+                userid=dict(required=False, type='str', default=None),
```
Comment (simon-spinner) [inline plugins/modules/zhmc_adapter.py:860]: As this the same for each module I wonder whether we could provide a common function create_hmc_auth_argument for that?

File: docs/source/modules/zhmc_adapter.rst (hunk 1 above)
Comment (andy-maier) [reply docs/source/modules/zhmc_adapter.rst:53]: I found to my surprise that the user name is required in both cases, i.e. also when session_id is used.

File: plugins/modules/zhmc_adapter.py (hunk 2 above)
Comment (andy-maier) [reply plugins/modules/zhmc_adapter.py:860]: If we do that, let's do it in a separate PR. If you think it is worth while, can you please open an issue for that?

File: docs/source/modules/zhmc_adapter.rst (hunk 1 above)
Comment (andy-maier) [reply docs/source/modules/zhmc_adapter.rst:53]: After discussion with Simon, it turns out that the HMC does not have a requirement to specify the userid if a session ID is specfied. So if the ibm_zhmc modules have that requirement, we need to find what that gets caused by, and remove the requirement.

Comment (andy-maier) [reply docs/source/modules/zhmc_adapter.rst:53]: The userid was not needed, and I updated the PR to not return it from the zhmc_session module anymore. As an input parameter to other modules, userid is now also mutually exclusive with session_id.
That should resolve the comment.

**Review by simon-spinner — APPROVED** [review]

**Pull request merged**